src/components/Stories.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+
+import api from '../services/api';
+import constant from '../constants/data';
+
+/**
+ * The main page showing all the top stories.
+ */
+class Stories extends React.Component {
+  /**
+   * Creates the instance of Home.
+   *
+   * @param {object} props
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      storyType: props.location.pathname,
+      storiesIdList: [],
+      stories: [],
+      currentPage: 0
+    };
+  }
+
+  /**
+   * Change the page number.
+   *
+   * @param {Number} value
+   */
+  pageHandler = value => {
+    if (
+      (this.state.currentPage === 0 && value === -1) ||
+      (this.state.currentPage === 4 && value === 1)
+    ) {
+      return;
+    }
+    this.setState(prevState => ({
+      currentPage: prevState.currentPage + value
+    }));
+  };
+
+  /**
+   * Load the storiesId and update StoriesIDList.
+   */
+  updateStoriesList = () => {
+    if (this.state.storiesIdList && this.state.storiesIdList.length > 1) {
+      this.loadStories();
+
+      return;
+    }
+    api(this.state.storyType)
```

Comment:
Don't use name like `api`. You're making a get request for something. Make it clear what you're trying to do with the name.